Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp

```diff
@@ -79,6 +81,9 @@ namespace AWSNativeSDKInit
         Platform::CustomizeSDKOptions(m_awsSDKOptions);
         Aws::InitAPI(m_awsSDKOptions);
 
+        //! Disable calls to EC2Metadata service by default
+        AZ::Utils::SetEnv("AWS_EC2_METADATA_DISABLED", "True", true);
```

Comment:
Another option would be to add a setting in the settings registry to control this value.  That way you could control it without compiling.

Reply:
Added a cvar to make toggling this on/off easier. May need to find new owner of fix, if we need to support setreg as well.